pkg/ipamd/ipamd.go

```diff
@@ -572,19 +571,9 @@ func (c *IPAMContext) nodeInit() error {
 		c.askForTrunkENIIfNeeded(ctx)
 	}
 
-	if !c.disableENIProvisioning {
+	if !c.disableENIProvisioning && c.isDatastorePoolTooLow() {
 		// For a new node, attach Cidrs (secondary ips/prefixes)
-		increasedPool, err := c.tryAssignCidrs()
-		if err == nil && increasedPool {
-			c.updateLastNodeIPPoolAction()
-		} else if err != nil {
-			if containsInsufficientCIDRsOrSubnetIPs(err) {
-				log.Errorf("Unable to attach IPs/Prefixes for the ENI, subnet doesn't seem to have enough IPs/Prefixes. Consider using new subnet or carve a reserved range using create-subnet-cidr-reservation")
-				c.lastInsufficientCidrError = time.Now()
-				return nil
-			}
-			return err
-		}
+		c.increaseDatastorePool(ctx)
```

Comment:
this slightly changed the behavior. previously an error will return if any thing wrong when allocating IPs and prevent node from became ready. e.g. missing permission.

but now it will just try to allocate IPs and silently swallow the error, and node will be ready any way.

shall we make c.increaseDatastorePool returns errors and return the error during bootstrap, and ignore the error(only logs it) during the IP pool reconcile loop?

Reply:
Yeah that right, it would be good to fail. Instead of modifying c.increaseDatastorePool where the code flow is slightly different i.e if the AssignCidr returns (non-nil && subnet has space) we go ahead with creating an ENI however here we need it to fail. I am thinking we can move this logic - https://github.com/jayanthvn/amazon-vpc-cni-k8s/blob/d753d02410137571afa50adb63897206627baf49/pkg/ipamd/ipamd.go#L794-L808 to a separate function and call it in nodeInit and increaseDataStorePool. Any thoughts?

Reply:
On a second thought, we can modify increaseDatastorePool so that we can re-use the same function. Will test it out.

Reply:
nvm, I restructured the code.

Reply:
Also, I am thinking to limit this for customer networking and the first eni. Will discuss more offline.

Reply:
Yes, would be good to limit this for Custom networking case. 

Instead of modifying `increaseDataStorePool` to return an error (and then ignore it everywhere else except at this location), I think we can maybe check if the call resulted in a change in the datastore contents (i.e.,) ENIs (and maybe IPs) count and use that to determine if the call succeeded or not and return an error if it didn't.